slices/perl-modules-5.38.yaml

```diff
@@ -0,0 +1,16 @@
+package: perl-modules-5.38
+
+essential:
+  - perl-modules-5.38_copyright
+
+slices:
+  modules:
```

Comment:
Here we have the additional architecture-irrelevant modules for perl. They can be roughly grouped by `CPAN`, 
 `carp`, `class` `compress`, `config`, `dbm_filter`, `devel`, `digest`, `encode`, `exporter`, `extutils`, `file`, `filter`, `getops`, `http`, `I18N`, `io`, `ipc`, `json`, `locale`, `math`, `memorize`, `module/...`, `net`, `params`, `parse`, `perl`, `perlio`, `pod`, `tap`, `term`, `test`, `test2`, `thread`, `tie`, `time`, `unicode`, `autodie`, `encoding`, `overload`, `unicore`, `warnings`, and many other files that are flat our there in `/usr/share/perl/5.38*`. I reckon that only `cpan`, `exutils`, `pod`, `tap` and `test*` are good to be extracted as standalone slices. We have no strong motivations to distinguish the other modules and make them into standalone slices.

Reply:
Then I reckon an issue: Since a large number of files and directories are scattered in the `/usr/share/perl/5.38*` directory, splitting them into slices (w/o overlapping and conflicts) will make the SDF extremely bloated.

Reply:
that's not an issue. See https://github.com/canonical/chisel-releases/blob/ubuntu-22.04/slices/openjdk-8-jre-headless.yaml and https://github.com/canonical/chisel-releases/blob/ubuntu-22.04/slices/libpython3.10-stdlib.yaml.

Doing "catch-all" slices is an anti pattern for chisel. With a bit of digging @linostar could potentially tell you which perl modules are useless for RabbitMQ for example, and then you'd reduce that image's attack surface even further.

Regardless of that, `perl` should be sliced with the same mindset as `python`, `java`...(`dotnet-runtime` should be done the same way)

This means a `core` slice may also be expected, for example, for `perl-base`, the `core` could be something like autoloader
carp
cwd
dynaloader
errorno
exporter
fcntl
file
hash
io
ipc
list
posix
scalar
socket
symbol
text
tie
xsloader
attributes
bytes
constant
feature
fields
integer
lib
locale
overload
parent
re
strict
utf8
vars
warnings

(?)

Reply:
I fully understand this mindset. However, I've discovered that even though we only have `python3.12-minimal`, many modules such as `socket`, `threading`, `os`, `argparse`, etc. are already included. This question then leaves: How should we define the scope of `perl-base` and `perl-modules`? What are the criteria that we determine whether a module should be a part of the `core` module?

Reply:
`python*-minimal` should also be further split. 

It is true that nested dependencies become then the challenge. 

In practice:

```yaml
A
slices:
  core:
    essential:
      - B_core

  functionality-X:
    essential:
       - A_core
       - B_functionality-X

---
B
slices:
  core:

  functionality-X:
    essential:
      - B_core
```